src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -369,7 +372,33 @@ public static void checkApiUrlValidity(@Nonnull GitHub gitHub, @CheckForNull Sta
             gb.withEndpoint(apiUrl);
             gb.withRateLimitHandler(CUSTOMIZED);
 
+
+            Interceptor interceptor = new Interceptor() {
+                @Override
+                public Response intercept(Chain chain) throws IOException {
+                    Request request = chain.request();
+
+                    // okhttp caches the tag response even if the tag object is not found.
+                    // tags are mutable, therefore we should not cache the references
+                    // https://issues.jenkins-ci.org/browse/JENKINS-55330
+                    Response originalResponse = chain.proceed(request);
+
+                    if (request.url().getPath().contains("git/refs/tags")) {
+                        return originalResponse.newBuilder()
+                            .removeHeader("Expires")
+                            .removeHeader("Pragma")
+                            .removeHeader("Cache-Control")
+                            .header("Cache-Control", "no-cache")
+                            .build();
+                    }
+                    return originalResponse;
+                }
+            };
+
             OkHttpClient client = new OkHttpClient().setProxy(getProxy(host));
+            // client.interceptors().add(interceptor);
+            client.networkInterceptors().add(interceptor);
+            client.setProxy(getProxy(host));
```

Comment:
The proxy is already configured three lines up

Reply:
thanks

Reply:
updated